### PR TITLE
sd added

### PR DESCRIPTION
**Pull request by chetak123**
added sd and a bit of README . sd found in b/*/release/sd

### CODE REVIEW

File: sd/install.sh [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    pkg_src="$pkg_src_cmd"
+
+    # pkg_install must be defined by every package
+    pkg_install() {
+        # ~/.local/opt/sd-v0.99.9/bin
+        mkdir -p "$(dirname $pkg_src_cmd)"
+
+        # mv ./sd-*/sd ~/.local/opt/sd-v0.99.9/bin/sd
+       # mv ./sd-*/sd "$pkg_src_cmd"
+       	mv b/*/release/sd "$pkg_src_cmd"
```
[the remark this answers is not in the document]
Comment (chetak123) [reply sd/install.sh:28]: yes: https://github.com/chmln/sd/releases/download/v0.7.5/sd.0.7.5-.x86_64-unknown-linux-gnu.zip
➜  Downloads unzip -l sd.0.7.5-.x86_64-unknown-linux-gnu.zip 
Archive:  sd.0.7.5-.x86_64-unknown-linux-gnu.zip
  Length      Date    Time    Name
---------  ---------- -----   ----
        0  2020-05-05 02:33   b/
        0  2020-05-05 02:33   b/release/
        0  2020-05-05 02:33   b/release/build/
        0  2020-05-05 02:33   b/release/build/sd-c11c8851503c29f5/
  3040176  2020-05-05 02:33   b/release/build/sd-c11c8851503c29f5/build-script-build
  3040176  2020-05-05 02:33   b/release/build/sd-c11c8851503c29f5/build_script_build-c11c8851503c29f5
        0  2020-05-05 02:33   b/release/build/proc-macro-error-97d7495b547bb94a/
       78  2020-05-05 02:33   b/release/build/proc-macro-error-97d7495b547bb94a/root-output
       48  2020-05-05 02:33   b/release/build/proc-macro-error-97d7495b547bb94a/invoked.timestamp
        0  2020-05-05 02:33   b/release/build/proc-macro-error-97d7495b547bb94a/stderr
       29  2020-05-05 02:33   b/release/build/proc-macro-error-97d7495b547bb94a/output
        0  2020-05-05 02:33   b/release/build/bitflags-b4a8a6d66ee92faa/
  2794040  2020-05-05 02:33   b/release/build/bitflags-b4a8a6d66ee92faa/build-script-build
  2794040  2020-05-05 02:33   b/release/build/bitflags-b4a8a6d66ee92faa/build_script_build-b4a8a6d66ee92faa
        0  2020-05-05 02:33   b/release/build/rayon-core-6bf8fc8be1a5a85a/
  2736544  2020-05-05 02:33   b/release/build/rayon-core-6bf8fc8be1a5a85a/build-script-build
  2736544  2020-05-05 02:33   b/release/build/rayon-core-6bf8fc8be1a5a85a/build_script_build-6bf8fc8be1a5a85a
        0  2020-05-05 02:33   b/release/build/proc-macro2-9fcedc9143981f0b/
  2809752  2020-05-05 02:33   b/release/build/proc-macro2-9fcedc9143981f0b/build_script_build-9fcedc9143981f0b
  2809752  2020-05-05 02:33   b/release/build/proc-macro2-9fcedc9143981f0b/build-script-build
        0  2020-05-05 02:33   b/release/build/memoffset-4eeb79fa89c80258/
  2831888  2020-05-05 02:33   b/release/build/memoffset-4eeb79fa89c80258/build-script-build
  2831888  2020-05-05 02:33   b/release/build/memoffset-4eeb79fa89c80258/build_script_build-4eeb79fa89c80258
        0  2020-05-05 02:33   b/release/build/crossbeam-utils-d644bbc7450a6c7c/
  2832272  2020-05-05 02:33   b/release/build/crossbeam-utils-d644bbc7450a6c7c/build_script_build-d644bbc7450a6c7c
  2832272  2020-05-05 02:33   b/release/build/crossbeam-utils-d644bbc7450a6c7c/build-script-build
        0  2020-05-05 02:33   b/release/build/proc-macro-error-attr-fdfccd27fae0b8aa/
       83  2020-05-05 02:33   b/release/build/proc-macro-error-attr-fdfccd27fae0b8aa/root-output
       48  2020-05-05 02:33   b/release/build/proc-macro-error-attr-fdfccd27fae0b8aa/invoked.timestamp
        0  2020-05-05 02:33   b/release/build/proc-macro-error-attr-fdfccd27fae0b8aa/stderr
        0  2020-05-05 02:33   b/release/build/proc-macro-error-attr-fdfccd27fae0b8aa/output
        0  2020-05-05 02:33   b/release/build/proc-macro2-c4e15ed31c7cff40/
       73  2020-05-05 02:33   b/release/build/proc-macro2-c4e15ed31c7cff40/root-output
       48  2020-05-05 02:33   b/release/build/proc-macro2-c4e15ed31c7cff40/invoked.timestamp
        0  2020-05-05 02:33   b/release/build/proc-macro2-c4e15ed31c7cff40/stderr
       95  2020-05-05 02:33   b/release/build/proc-macro2-c4e15ed31c7cff40/output
        0  2020-05-05 02:33   b/release/build/maybe-uninit-5c99368adb3960a6/
  2793936  2020-05-05 02:33   b/release/build/maybe-uninit-5c99368adb3960a6/build-script-build
  2793936  2020-05-05 02:33   b/release/build/maybe-uninit-5c99368adb3960a6/build_script_build-5c99368adb3960a6
        0  2020-05-05 02:33   b/release/build/proc-macro-error-attr-74f93f22a630019b/
  2808840  2020-05-05 02:33   b/release/build/proc-macro-error-attr-74f93f22a630019b/build-script-build
  2808840  2020-05-05 02:33   b/release/build/proc-macro-error-attr-74f93f22a630019b/build_script_build-74f93f22a630019b
        0  2020-05-05 02:33   b/release/build/syn-83820677e84fd469/
       65  2020-05-05 02:33   b/release/build/syn-83820677e84fd469/root-output
       48  2020-05-05 02:33   b/release/build/syn-83820677e84fd469/invoked.timestamp
        0  2020-05-05 02:33   b/release/build/syn-83820677e84fd469/stderr
       42  2020-05-05 02:33   b/release/build/syn-83820677e84fd469/output
        0  2020-05-05 02:33   b/release/build/getrandom-f443270e627ad8f4/
  2741472  2020-05-05 02:33   b/release/build/getrandom-f443270e627ad8f4/build_script_build-f443270e627ad8f4
  2741472  2020-05-05 02:33   b/release/build/getrandom-f443270e627ad8f4/build-script-build
        0  2020-05-05 02:33   b/release/build/memchr-c2839981da5f7cfd/
  2746472  2020-05-05 02:33   b/release/build/memchr-c2839981da5f7cfd/build-script-build
  2746472  2020-05-05 02:33   b/release/build/memchr-c2839981da5f7cfd/build_script_build-c2839981da5f7cfd
        0  2020-05-05 02:33   b/release/build/bitflags-a957b330273a8ee6/
       70  2020-05-05 02:33   b/release/build/bitflags-a957b330273a8ee6/root-output
       48  2020-05-05 02:33   b/release/build/bitflags-a957b330273a8ee6/invoked.timestamp
        0  2020-05-05 02:33   b/release/build/bitflags-a957b330273a8ee6/stderr
       34  2020-05-05 02:33   b/release/build/bitflags-a957b330273a8ee6/output
        0  2020-05-05 02:33   b/release/build/libc-669d99a268e3183f/
       66  2020-05-05 02:33   b/release/build/libc-669d99a268e3183f/root-output
       48  2020-05-05 02:33   b/release/build/libc-669d99a268e3183f/invoked.timestamp
        0  2020-05-05 02:33   b/release/build/libc-669d99a268e3183f/stderr
      210  2020-05-05 02:33   b/release/build/libc-669d99a268e3183f/output
        0  2020-05-05 02:33   b/release/build/libc-dd66fc4c8b9492bc/
  2798888  2020-05-05 02:33   b/release/build/libc-dd66fc4c8b9492bc/build-script-build
  2798888  2020-05-05 02:33   b/release/build/libc-dd66fc4c8b9492bc/build_script_build-dd66fc4c8b9492bc
        0  2020-05-05 02:33   b/release/build/proc-macro-error-eec5a070650a8d5b/
  2809184  2020-05-05 02:33   b/release/build/proc-macro-error-eec5a070650a8d5b/build_script_build-eec5a070650a8d5b
  2809184  2020-05-05 02:33   b/release/build/proc-macro-error-eec5a070650a8d5b/build-script-build
        0  2020-05-05 02:33   b/release/build/syn-9a99fceb3158e59b/
  2794216  2020-05-05 02:33   b/release/build/syn-9a99fceb3158e59b/build-script-build
  2794216  2020-05-05 02:33   b/release/build/syn-9a99fceb3158e59b/build_script_build-9a99fceb3158e59b
        0  2020-05-05 02:33   b/release/build/crossbeam-epoch-4669bc310aad8b3f/
  2831888  2020-05-05 02:33   b/release/build/crossbeam-epoch-4669bc310aad8b3f/build-script-build
  2831888  2020-05-05 02:33   b/release/build/crossbeam-epoch-4669bc310aad8b3f/build_script_build-4669bc310aad8b3f
        0  2020-05-05 02:33   b/x86_64-unknown-linux-gnu/
        0  2020-05-05 02:33   b/x86_64-unknown-linux-gnu/release/
        0  2020-05-05 02:33   b/x86_64-unknown-linux-gnu/release/build/
        0  2020-05-05 02:33   b/x86_64-unknown-linux-gnu/release/build/sd-bb15ef4d4286b733/
        0  2020-05-05 02:33   b/x86_64-unknown-linux-gnu/release/build/sd-bb15ef4d4286b733/out/
     1088  2020-05-05 02:33   b/x86_64-unknown-linux-gnu/release/build/sd-bb15ef4d4286b733/out/sd.elv
     1334  2020-05-05 02:33   b/x86_64-unknown-linux-gnu/release/build/sd-bb15ef4d4286b733/out/sd.1
     1234  2020-05-05 02:33   b/x86_64-unknown-linux-gnu/release/build/sd-bb15ef4d4286b733/out/sd.bash
     1286  2020-05-05 02:33   b/x86_64-unknown-linux-gnu/release/build/sd-bb15ef4d4286b733/out/_sd
     2216  2020-05-05 02:33   b/x86_64-unknown-linux-gnu/release/build/sd-bb15ef4d4286b733/out/_sd.ps1
      499  2020-05-05 02:33   b/x86_64-unknown-linux-gnu/release/build/sd-bb15ef4d4286b733/out/sd.fish
       89  2020-05-05 02:33   b/x86_64-unknown-linux-gnu/release/build/sd-bb15ef4d4286b733/root-output
       48  2020-05-05 02:33   b/x86_64-unknown-linux-gnu/release/build/sd-bb15ef4d4286b733/invoked.timestamp
        0  2020-05-05 02:33   b/x86_64-unknown-linux-gnu/release/build/sd-bb15ef4d4286b733/stderr
        0  2020-05-05 02:33   b/x86_64-unknown-linux-gnu/release/build/sd-bb15ef4d4286b733/output
        0  2020-05-05 02:33   b/x86_64-unknown-linux-gnu/release/build/memoffset-0c43baafe0a4fa15/
        0  2020-05-05 02:33   b/x86_64-unknown-linux-gnu/release/build/memoffset-0c43baafe0a4fa15/out/
      287  2020-05-05 02:33   b/x86_64-unknown-linux-gnu/release/build/memoffset-0c43baafe0a4fa15/out/probe0.ll
       96  2020-05-05 02:33   b/x86_64-unknown-linux-gnu/release/build/memoffset-0c43baafe0a4fa15/root-output
       48  2020-05-05 02:33   b/x86_64-unknown-linux-gnu/release/build/memoffset-0c43baafe0a4fa15/invoked.timestamp
        0  2020-05-05 02:33   b/x86_64-unknown-linux-gnu/release/build/memoffset-0c43baafe0a4fa15/stderr
       83  2020-05-05 02:33   b/x86_64-unknown-linux-gnu/release/build/memoffset-0c43baafe0a4fa15/output
        0  2020-05-05 02:33   b/x86_64-unknown-linux-gnu/release/build/getrandom-4c061e11a423413c/
       96  2020-05-05 02:33   b/x86_64-unknown-linux-gnu/release/build/getrandom-4c061e11a423413c/root-output
       48  2020-05-05 02:33   b/x86_64-unknown-linux-gnu/release/build/getrandom-4c061e11a423413c/invoked.timestamp
        0  2020-05-05 02:33   b/x86_64-unknown-linux-gnu/release/build/getrandom-4c061e11a423413c/stderr
        0  2020-05-05 02:33   b/x86_64-unknown-linux-gnu/release/build/getrandom-4c061e11a423413c/output
        0  2020-05-05 02:33   b/x86_64-unknown-linux-gnu/release/build/libc-fcabc6ee628fded4/
       91  2020-05-05 02:33   b/x86_64-unknown-linux-gnu/release/build/libc-fcabc6ee628fded4/root-output
       48  2020-05-05 02:33   b/x86_64-unknown-linux-gnu/release/build/libc-fcabc6ee628fded4/invoked.timestamp
        0  2020-05-05 02:33   b/x86_64-unknown-linux-gnu/release/build/libc-fcabc6ee628fded4/stderr
      210  2020-05-05 02:33   b/x86_64-unknown-linux-gnu/release/build/libc-fcabc6ee628fded4/output
        0  2020-05-05 02:33   b/x86_64-unknown-linux-gnu/release/build/memchr-699eb62edde13d69/
       93  2020-05-05 02:33   b/x86_64-unknown-linux-gnu/release/build/memchr-699eb62edde13d69/root-output
       48  2020-05-05 02:33   b/x86_64-unknown-linux-gnu/release/build/memchr-699eb62edde13d69/invoked.timestamp
        0  2020-05-05 02:33   b/x86_64-unknown-linux-gnu/release/build/memchr-699eb62edde13d69/stderr
      144  2020-05-05 02:33   b/x86_64-unknown-linux-gnu/release/build/memchr-699eb62edde13d69/output
        0  2020-05-05 02:33   b/x86_64-unknown-linux-gnu/release/build/maybe-uninit-ebd3b0f9d7d6db95/
       99  2020-05-05 02:33   b/x86_64-unknown-linux-gnu/release/build/maybe-uninit-ebd3b0f9d7d6db95/root-output
       48  2020-05-05 02:33   b/x86_64-unknown-linux-gnu/release/build/maybe-uninit-ebd3b0f9d7d6db95/invoked.timestamp
        0  2020-05-05 02:33   b/x86_64-unknown-linux-gnu/release/build/maybe-uninit-ebd3b0f9d7d6db95/stderr
       91  2020-05-05 02:33   b/x86_64-unknown-linux-gnu/release/build/maybe-uninit-ebd3b0f9d7d6db95/output
        0  2020-05-05 02:33   b/x86_64-unknown-linux-gnu/release/build/bitflags-0826a46429fd82aa/
       95  2020-05-05 02:33   b/x86_64-unknown-linux-gnu/release/build/bitflags-0826a46429fd82aa/root-output
       48  2020-05-05 02:33   b/x86_64-unknown-linux-gnu/release/build/bitflags-0826a46429fd82aa/invoked.timestamp
        0  2020-05-05 02:33   b/x86_64-unknown-linux-gnu/release/build/bitflags-0826a46429fd82aa/stderr
       34  2020-05-05 02:33   b/x86_64-unknown-linux-gnu/release/build/bitflags-0826a46429fd82aa/output
        0  2020-05-05 02:33   b/x86_64-unknown-linux-gnu/release/build/crossbeam-epoch-72fcfe788c606437/
        0  2020-05-05 02:33   b/x86_64-unknown-linux-gnu/release/build/crossbeam-epoch-72fcfe788c606437/out/
      287  2020-05-05 02:33   b/x86_64-unknown-linux-gnu/release/build/crossbeam-epoch-72fcfe788c606437/out/probe0.ll
      102  2020-05-05 02:33   b/x86_64-unknown-linux-gnu/release/build/crossbeam-epoch-72fcfe788c606437/root-output
       48  2020-05-05 02:33   b/x86_64-unknown-linux-gnu/release/build/crossbeam-epoch-72fcfe788c606437/invoked.timestamp
        0  2020-05-05 02:33   b/x86_64-unknown-linux-gnu/release/build/crossbeam-epoch-72fcfe788c606437/stderr
       33  2020-05-05 02:33   b/x86_64-unknown-linux-gnu/release/build/crossbeam-epoch-72fcfe788c606437/output
        0  2020-05-05 02:33   b/x86_64-unknown-linux-gnu/release/build/crossbeam-utils-8e604eace3fc5fb5/
        0  2020-05-05 02:33   b/x86_64-unknown-linux-gnu/release/build/crossbeam-utils-8e604eace3fc5fb5/out/
      287  2020-05-05 02:33   b/x86_64-unknown-linux-gnu/release/build/crossbeam-utils-8e604eace3fc5fb5/out/probe3.ll
      287  2020-05-05 02:33   b/x86_64-unknown-linux-gnu/release/build/crossbeam-utils-8e604eace3fc5fb5/out/probe1.ll
      287  2020-05-05 02:33   b/x86_64-unknown-linux-gnu/release/build/crossbeam-utils-8e604eace3fc5fb5/out/probe4.ll
      287  2020-05-05 02:33   b/x86_64-unknown-linux-gnu/release/build/crossbeam-utils-8e604eace3fc5fb5/out/probe2.ll
      287  2020-05-05 02:33   b/x86_64-unknown-linux-gnu/release/build/crossbeam-utils-8e604eace3fc5fb5/out/probe0.ll
      102  2020-05-05 02:33   b/x86_64-unknown-linux-gnu/release/build/crossbeam-utils-8e604eace3fc5fb5/root-output
       48  2020-05-05 02:33   b/x86_64-unknown-linux-gnu/release/build/crossbeam-utils-8e604eace3fc5fb5/invoked.timestamp
      380  2020-05-05 02:33   b/x86_64-unknown-linux-gnu/release/build/crossbeam-utils-8e604eace3fc5fb5/stderr
      156  2020-05-05 02:33   b/x86_64-unknown-linux-gnu/release/build/crossbeam-utils-8e604eace3fc5fb5/output
        0  2020-05-05 02:33   b/x86_64-unknown-linux-gnu/release/build/rayon-core-79d81673c655b045/
       97  2020-05-05 02:33   b/x86_64-unknown-linux-gnu/release/build/rayon-core-79d81673c655b045/root-output
       48  2020-05-05 02:33   b/x86_64-unknown-linux-gnu/release/build/rayon-core-79d81673c655b045/invoked.timestamp
        0  2020-05-05 02:33   b/x86_64-unknown-linux-gnu/release/build/rayon-core-79d81673c655b045/stderr
       32  2020-05-05 02:33   b/x86_64-unknown-linux-gnu/release/build/rayon-core-79d81673c655b045/output
  2890896  2020-05-05 02:33   b/x86_64-unknown-linux-gnu/release/sd
---------                     -------
 81643434                     146 files

**Comment by solderjs** [comment]
I was able to test this on Linux and, surprisingly, it worked despite not working on Mac.

It looks like `sd` has recently changed their release format ( https://github.com/chmln/sd/issues/97 ) and that's tripping up the installer logic.

File: sd/README.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+Use the `@beta` tag for pre-releases.
+
+
+## Cheat Sheet
+
+> sd - s[earch] & d[isplace]
+> sd uses regex syntax that you already know from JavaScript and Python. Forget about dealing with quirks of sed or awk - get productive immediately.
+
+
```
[the remark this answers is not in the document]
Comment (solderjs) [reply sd/README.md:20]: Please remove this entire section that was copied from the official README.md. The cheat sheet is intended to be original content that showcases the best parts of the original tool, but not to copy or potentially cause confusion with the official source repo.

If you don't have any examples that you can add from you experience with `sd`, I'll add some later.

I think that the new example that you gave is great:

```
sd 'original word' 'final word' file_to_be_changed
```

That's the sort of thing that we want.

**Comment by solderjs** [comment]
Do you know about `git rebase -i`?

To remove the files from the git history so that they are not in the merge you can use `git rebase -i master` and then select to `squash` all of the intermediary commits and this will keep the merge clean and it will not add the extra files to the repository.

**Comment by solderjs** [comment]
@chetak123 It looks like I forgot to update the test to check for non-packaged `exe` and `bin` files: https://github.com/webinstall/packages/issues/89

This is the reason that it worked on Linux when you tested, but was not working for me on Mac.

It looks like we will need to have the Mac / Linux install.sh use the new release format (no zip file, just the binary), and the Windows install.ps1 will have to support the old format for now.

**Comment by chetak123** [comment]
squashed the 8 commits to a single commit

File: sd/README.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+To run sd:
+
+
+1. **Find & replace in a file**
+
+```sh
+> sd 'original word' 'final word' file_to_be_changed
+```
+
+## Comparison to sed
```
Comment (solderjs) [inline sd/README.md:30]: Please remove this copied section and re-squash.

**Comment by solderjs** [comment]
@chetak123 That's great. There's just one thing that I think has been really unclear that needs to be fixed:

We don't want to directly copy content from the official README, except for the tagline and perhaps the description.

The `Cheat Sheet` section is intended to be original content only.

- [ ] Remove all copied content from Cheat Sheet
- [ ] Add some original content (i.e. `sd 'original word' 'replacement word' ./file_to_be_changed`)
- [ ] Re-squash commits
- [ ] `git pull --rebase https://github.com/webinstall/packages master`
- [ ] `git push --force <REMOTE-NAME> master`

I'm sorry for the miscommunication. Thanks for your patience.

File: _webi/test.js [hunk 4]
```diff
@@ -66,7 +66,7 @@ Releases.get(path.join(process.cwd(), pkgdir)).then(function (all) {
   var pkgname = path.basename(pkgdir.replace(/\/$/, ''));
```
Comment (solderjs) [inline _webi/test.js:66]: This commit is already merged into master and shouldn't show up in your list. I wonder what went wrong here.

**Comment by chetak123** [comment]
added some examples

File: _webi/test.js (hunk 4 above)
Comment (chetak123) [reply _webi/test.js:66]: i don't know how but after i did  git pull --rebase https://github.com/webinstall/packages master its showing me recent commits made by other people along with my commit

**Comment by solderjs** [comment]
@chetak123 This is looking really good, there are just two issues:

- [ ] It looks like you squashed a commit that wasn't yours by mistake.
    - `git pull --rebase https://github.com/webinstall/packages master` should fix that
    - also, could you clean up that commit message to just "add sd" rather than having dozens of lines messages from the prior commits? This is not a blocker, but it would be nice.
- [ ] The README is NOT rendering as proper Markdown
    - Code blocks are missing the code fences:
    ~~~md
    ```bash
    # code here ...
    ```
    ~~~
    - The examples have these extra invalid characters: `➜  `
    - The heading for Cheat Sheet was removed. We do want the Cheat Sheet heading.

For reference, this is the format the README should follow: https://webinstall.dev/comrak/

File: _webi/test.js (hunk 4 above)
Comment (solderjs) [reply _webi/test.js:66]: That's what rebase does (it brings you up-to-date and the replays your new commits on top of what's in the most current master). Just rebase again and don't squash those commits this time.

Please also see my other comments in issue comments.

**Comment by chetak123** [comment]
made some changes for #96 , squashed them to single commit

**Comment by solderjs** [comment]
Something is still wrong with the rebase (you're not the author of your own commit), but I merged and I've got this up on https://beta.webinstall.dev/sd

# Doesn't work on Mac and Linux:

```bash
curl https://beta.webinstall.dev/sd | bash
```

And you'll get an error:

```txt
mv: rename b/*/release/sd to /Users/aj/.local/opt/sd-v0.7.6/bin/sd: No such file or directory
```

The latest `0.7.6` release for Mac and Linux is actually flat binaries, not nested in a `.zip`.

Here's the change I made to get it to work:

```bash
        # mv ./sd-* ~/.local/opt/sd-v0.99.9/bin/sd
        mv sd-* "$pkg_src_cmd"
```

# Doesn't work on Windows

```pwsh
curl.exe -A "MS" https://beta.webinstall.dev/sd | powershell
```

Here's the change I made to get it to work:

```pwsh
        # Remove any leftover tmp cruft
        Remove-Item -Path ".\sd-*" -Recurse -ErrorAction Ignore
        Remove-Item -Path ".\sd.exe" -Recurse -ErrorAction Ignore

        # Unpack archive file into this temporary directory
        # Windows BSD-tar handles zip. Imagine that.
        echo "Unpacking $pkg_download"
        & tar xf "$pkg_download"
        & dir

        # Settle unpacked archive into place
        echo "Install Location: $pkg_src_cmd"
        New-Item "$pkg_src_bin" -ItemType Directory -Force
        Move-Item -Path "b\*\release\sd.exe" -Destination "$pkg_src_bin"
```

**Comment by solderjs** [comment]
Before the next time you commit do `git reset e40f82e`, which is the exact commit of the current `master`.

**Comment by coolaj86** [comment]
@chetak123 Are you still working on this? I've tested the changes I suggested, so I know that they will work.

**Comment by chetak123** [comment]
when i download sd , its not executable , so i have to manually type chmod +x ~/.local/opt/sd-*/bin/sd, where should i put it in script?

File: sd/install.sh [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    pkg_src_cmd="$HOME/.local/opt/sd-v$WEBI_VERSION/bin/sd"
+    pkg_src_dir="$HOME/.local/opt/sd-v$WEBI_VERSION"
+    pkg_src="$pkg_src_cmd"
+
+    # pkg_install must be defined by every package
+    pkg_install() {
+        # ~/.local/opt/sd-v0.99.9/bin
+        mkdir -p "$(dirname $pkg_src_cmd)"
+
+        # mv ./sd-*/sd ~/.local/opt/sd-v0.99.9/bin/sd
```
Comment (coolaj86) [inline sd/install.sh:26]: Got some step-ladder spacing going on here...

File: sd/install.ps1 [hunk 6]
```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env pwsh
+
+##################
+# Install sd #
+##################
+
```
Comment (coolaj86) [inline sd/install.ps1:6]: This is missing `WEBI_SINGLE=true`... perhaps the example is missing it too. Oops.

Comment (coolaj86) [reply sd/install.ps1:6]: Indeed, it is missing from `_example/install.sh` also. But it's in all of the other packages that need it, such as `rg/install.sh`. You can see what it looks like there.

**Comment by coolaj86** [comment]
@chetak123 It's already in there: https://github.com/webinstall/packages/blob/master/_webi/template.sh#L229

It just wasn't being run because one of the variables that is supposed to be set `WEBI_SINGLE=true` was missing.

File: sd/install.ps1 (hunk 6 above)
Comment (chetak123) [reply sd/install.ps1:6]: WEBI_SINGLE=true is not in rg/install.sh https://github.com/webinstall/packages/blob/master/rg/install.sh

**Comment by coolaj86** [comment]
I opened a bug to address the `chmod` being in the wrong place:
https://github.com/webinstall/packages/issues/100

The goal is to get rid of `WEBI_SINGLE`, so better to fix the `chmod` than add `WEBI_SINGLE` where it otherwise won't be needed.

**Comment by chetak123** [comment]
updated to latest commit

**Review by coolaj86 — APPROVED** [review]

**Comment by coolaj86** [comment]
I'm getting a conflict when I try to rebase this. Will you try this, please?

```bash
git reset b4915c67af11dea2ff0df9aaaa340bc375655e27
git stash
git pull --rebase https://github.com/webinstall/packages master
git stash pop
git add sd/
git commit -m "add sd"
git push --force origin master
```

**Pull request merged**